src/gfx/2d/surface.ts

```diff
@@ -308,9 +325,43 @@ module Shumway.GFX.Canvas2D {
         this.context.clip();
       }
       this.context.globalCompositeOperation = getCompositeOperation(blendMode);
-      this.context.drawImage(sourceCanvas, sx, sy, w, h, x, y, w, h);
+      Filters._applyColorMatrix(this.context, colorMatrix);
+      if (filters && filters.length > 1) {
+        filters = filters.slice();
+        var dx, dy, _cc, _sx, _sy;
+        if (copyContext) {
+          _cc = copyContext;
+          copyContext = sourceContext;
+          sourceContext = _cc;
+        } else {
+          Canvas2DSurfaceRegion._ensureCopyCanvasSize(w, h);
+          copyContext = Canvas2DSurfaceRegion._copyCanvasContext;
+          dx = 0;
+          dy = 0;
+        }
+        var filter = [null];
```

Comment:
Creates extra array objects. Could you extract getBlurScale (move outside of class) and _applySvgFilter logic from _applyFilters method? Adding e.g. _applyFilter will avoid creation of the intermediate arrays.